gtsfm/utils/triangulation.py

```diff
@@ -0,0 +1,47 @@
+"""Utilities to support triangulation.
+
+Authors: Ayush Baid
+"""
+import numpy as np
+from gtsam import PinholeCameraCal3Bundler
+
+
+def calculate_triangulation_angle_in_degrees(
+    camera_1: PinholeCameraCal3Bundler, camera_2: PinholeCameraCal3Bundler, point_3d: np.ndarray
+) -> float:
+    """Calculates the angle formed at the 3D point by the rays backprojected from 2 cameras.
+    In the setup with X (point_3d) and two cameras C1 and C2, the triangulation angle is the angle between rays C1-X and C2-X, i.e. the angle subtendted at the 3d point.1
+        X
+       / \
+      /   \
+     /     \
+    C1      C2
+
+    Args:
+        camera_1: the first camera.
+        camera_2: the second camera.
+        point_3d: the 3d point to compute the angle at.
+
+    Returns:
+        the angle formed at the 3d point, in degrees.
+    """
+    camera_center_1: np.ndarray = camera_1.pose().translation()
+    camera_center_2: np.ndarray = camera_2.pose().translation()
+
+    # compute the three squared edge lengths of the triangle formed between camera centers and the 3d point
+    def squared_dist_fn(x, y):
+        return np.sum(np.square(x - y), axis=None)
+
+    baseline_squared = squared_dist_fn(camera_center_1, camera_center_2)
```

Comment:
isn't it simpler to just create two vectors (from point to each camera), and compute the angle between them using the typical dot product cosine equation?

a \dot b = |a| |b| cos \theta

We already have that function implemented here: https://github.com/borglab/gtsfm/blob/master/gtsfm/utils/geometry_comparisons.py#L281

Reply:
yep makes sense. Thanks!